conjure-go-client/httpclient/authn.go

```diff
@@ -59,3 +60,25 @@ func newAuthTokenMiddlewareFromRefreshable(token refreshable.StringPtr) Middlewa
 		}},
 	}
 }
+
+// basicAuthMiddleware wraps a refreshing BasicAuth pointer and injects basic auth credentials if the pointer is not nil
+type basicAuthMiddleware struct {
+	basicAuth refreshingclient.RefreshableBasicAuthPtr
```

Comment:
To provide feature parity with auth tokens, could you create a `BasicAuthProvider` similar to the `TokenProvider`, add a `WithBasicAuthProvider` option to the HTTP client and switch the existing `WithBasicAuth` option to delegate to  `WithBasicAuthProvider` ?